packages/ui-components/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { usePopper } from "react-popper";
+import classNames from "classnames/bind";
+import css from "./Tooltip.module.scss";
+
+type TooltipPosition =
+  | "left"
+  | "left-start"
+  | "left-end"
+  | "right"
+  | "right-start"
+  | "right-end"
+  | "top"
+  | "top-start"
+  | "top-end"
+  | "bottom"
+  | "bottom-start"
+  | "bottom-end";
+
+type TooltipStyle = "default" | "light" | "dark";
+
+interface TooltipProps {
+  placement?: TooltipPosition;
+  style?: TooltipStyle;
+  content: ReactElement | string;
+  visible?: boolean;
+}
+
+const Tooltip: FunctionComponent<TooltipProps> = ({
+  placement = "bottom",
+  style = "default",
+  children,
+  content,
+  visible = false,
+}) => {
+  const [referenceElement, setReferenceElement] = useState(null);
+  const [popperElement, setPopperElement] = useState(null);
+  const [arrowElement, setArrowElement] = useState(null);
+
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    placement,
+    modifiers: [
+      { name: "arrow", options: { element: arrowElement } },
+      { name: "offset", options: { offset: [0, 10] } },
+      {
+        name: "computeStyles",
+        options: {
+          gpuAcceleration: false, // true by default
+        },
+      },
+    ],
+  });
+
+  if (!content) {
+    return <>{children}</>;
+  }
+
+  const classnames = classNames(css[style], css.Tooltip);
+  if (visible && popperElement) {
+    popperElement.setAttribute("data-show", "");
+  }
+
+  const wrappedChildren = React.Children.map(children, child => {
+    return React.cloneElement(
+      typeof children === "string" ? (
+        <span>{child}</span>
+      ) : (
+        (child as ReactElement)
+      ),
+      {
+        ref: setReferenceElement,
+        onMouseOver: () => {
+          popperElement.setAttribute("data-show", "");
+        },
+        onMouseOut: () => {
+          if (visible) return;
+          popperElement.removeAttribute("data-show");
+        },
+      },
+    );
+  });
+
+  return (
+    <>
+      {wrappedChildren}
+
+      <div
+        className={classnames}
+        ref={setPopperElement}
+        style={styles.popper}
+        data-jest="tooltip"
+        {...attributes.popper}
+      >
+        {content}
+        <div className={css.arrow} ref={setArrowElement} style={styles.arrow} />
+        <div className={css.arrowInner} style={styles.arrow} />
```

Comment:
Tell me about why we need this `arrowInner`. I'm playing around with it in the browser and I can tell what difference it makes if it's there or not.

Reply:
this needed if border of tooltip differs in color from background. for example, here is a `light` style without `arrowInner`:
![Screenshot 2020-05-27 at 13 46 49](https://user-images.githubusercontent.com/12850886/83010361-d3bd0800-a020-11ea-858c-a62690b54db3.png)